vispy/visuals/volume.py

```diff
@@ -497,23 +525,19 @@
                     gl_FragColor = applyColormap(color.r);
 
                     // set the variables for the depth buffer
-                    surface_point = iloc * u_shape;
-                    surface_found = true;
+                    frag_depth_point = iloc * u_shape;
```

Comment:
Instead of juggling around the variable, I made it more explicit: the `discard `happens below in the `else` block. Though I realize now that maybe it should be one level deeper, cause we want to be able to discard fragments that passed the first coarse test?

Reply:
Oh wait, this is the refinement stage. In that case I think its fine, because I don't think its possible/likely to hit the threshold in the coarse test and not in the refinement.

Reply:
I guess it can happen if you pass by parallel to a surface from within 0.2 distrance...

Reply:
In that case the values for the coarse location would be used, right?

Reply:
Mhm... In that case the fragment would not be discarded, and it's depth would stay at the default, which is at the closest face of the cube. Also, it's color would be unset, and thus transparent. This can be a problem with transparency issues, so I think we should solve it.

Wait, actually it's all wrong here, cause I'm discarding *in the loop*. Thius should happen *after the loop* anyways o.o